src/marks/difference.js

```diff
@@ -1,143 +1,155 @@
-import {area as shapeArea, line as shapeLine} from "d3";
+import {area as shapeArea} from "d3";
 import {create} from "../context.js";
-import {maybeCurve} from "../curve.js";
-import {Mark, withTip} from "../mark.js";
-import {identity, indexOf, isColor, number} from "../options.js";
-import {applyIndirectStyles, applyTransform, getClipId, groupIndex} from "../style.js";
+import {identity, indexOf, column, valueof} from "../options.js";
+import {basic as transform} from "../transforms/basic.js";
+import {groupIndex, getClipId} from "../style.js";
+import {marks} from "../mark.js";
+import {area} from "./area.js";
+import {lineY} from "./line.js";
 
-const defaults = {
-  ariaLabel: "difference",
-  fill: "none",
-  stroke: "currentColor",
-  strokeWidth: 1.5,
-  strokeLinecap: "round",
-  strokeLinejoin: "round",
-  strokeMiterlimit: 1
-};
-
-function maybeColor(value) {
-  if (value == null) return "none";
-  if (!isColor(value)) throw new Error(`invalid color: ${value}`);
-  return value;
+function renderArea(X, Y, y0, {curve}) {
+  return shapeArea()
+    .curve(curve)
+    .defined((i) => i >= 0) // TODO: ??
```

Comment:
The -1 is populated by the groupIndex helper, which is used by the area and line marks:

https://github.com/observablehq/plot/blob/7098bf3634f80442cca7efb626e275ca839599d3/src/style.js#L265-L271

```suggestion
    .defined((i) => i >= 0)
```